src/msgpack/FloatConverter.cs

```diff
@@ -0,0 +1,184 @@
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+
+namespace TarantoolDnx.MsgPack
+{
+    internal class FloatConverter : IMsgPackConverter<float>, IMsgPackConverter<double>
+    {
+        public void Write(float value, Stream stream, MsgPackSettings settings)
+        {
+            var binary = new FloatBinary(value);
+            stream.WriteByte((byte) DataTypes.Single);
+            if (BitConverter.IsLittleEndian)
+            {
+                stream.WriteByte(binary.byte3);
+                stream.WriteByte(binary.byte2);
+                stream.WriteByte(binary.byte1);
+                stream.WriteByte(binary.byte0);
+            }
+            else
+            {
+                stream.WriteByte(binary.byte0);
+                stream.WriteByte(binary.byte1);
+                stream.WriteByte(binary.byte2);
+                stream.WriteByte(binary.byte3);
+            }
+        }
+
+        float IMsgPackConverter<float>.Read(Stream stream, MsgPackSettings settings, Func<float> creator)
+        {
+            var type = (DataTypes)stream.ReadByte();
+
+            if (type != DataTypes.Single)
+                throw ExceptionUtils.BadTypeException(type, DataTypes.Single);
+
+            return ReadFloat(stream);
+        }
+
+        public void Write(double value, Stream stream, MsgPackSettings settings)
+        {
+            var binary = new DoubleBinary(value);
+            stream.WriteByte((byte) DataTypes.Double);
+            if (BitConverter.IsLittleEndian)
+            {
+                stream.WriteByte(binary.byte7);
+                stream.WriteByte(binary.byte6);
+                stream.WriteByte(binary.byte5);
+                stream.WriteByte(binary.byte4);
+                stream.WriteByte(binary.byte3);
+                stream.WriteByte(binary.byte2);
+                stream.WriteByte(binary.byte1);
+                stream.WriteByte(binary.byte0);
+            }
+            else
+            {
+                stream.WriteByte(binary.byte0);
+                stream.WriteByte(binary.byte1);
+                stream.WriteByte(binary.byte2);
+                stream.WriteByte(binary.byte3);
+                stream.WriteByte(binary.byte4);
+                stream.WriteByte(binary.byte5);
+                stream.WriteByte(binary.byte6);
+                stream.WriteByte(binary.byte7);
+            }
+        }
+
+        double IMsgPackConverter<double>.Read(Stream stream, MsgPackSettings settings, Func<double> creator)
```

Comment:
I mean why Write method implemented implicit, but Read method - explicit?
Why not to use:
`public double Read(Stream stream, MsgPackSettings settings, Func<double> creator)`
?